src/App.tsx

```diff
@@ -1,93 +1,78 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useState, useRef, useEffect } from 'react';
+import { useLocalStorage } from './hooks/useLocalStorage';
+import { Todo } from './types/Todo';
+import { FilterOptions } from './types/FilterOptions';
+import { TodosContext } from './store/TodosContext';
+import { TodoList } from './components/TodoList';
+import { TodoFilter } from './components/TodoFilter';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useLocalStorage<Todo[]>('todos', []);
+  const [newTitle, setNewTitle] = useState('');
+  const [filter, setFilter] = useState(FilterOptions.All);
+  const inputRef = useRef<HTMLInputElement | null>(null);
+
+  const addTodo = () => {
+    const newTodo = {
+      id: +new Date(),
+      title: newTitle,
+      completed: false,
+    };
+
+    setTodos([...todos, newTodo]);
```

Comment:
always use callback in state if you need to set new value based on the previous one